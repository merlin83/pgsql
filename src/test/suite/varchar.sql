---------------------------------------------------------------------------
--
-- varchar.sql-
--    test CHAR() and VARCHAR() adts
--
--
-- Copyright (c) 1994-5, Regents of the University of California
--
-- $Id: varchar.sql,v 1.1 1996-07-09 06:22:30 scrappy Exp $
--
---------------------------------------------------------------------------

-- test char(): insert w/ boundary cases
create table f (x char(5));
insert into f values ('zoo');
insert into f values ('a');
insert into f values ('jet');
insert into f values ('abc');
insert into f values ('');
insert into f values ('a c');
insert into f values ('abxyzxyz');
select * from f;
select * from f where x = 'jet';
select * from f where x <> 'jet';
select * from f where x < 'jet';
select * from f where x <= 'jet';
select * from f where x > 'jet';
select * from f where x >= 'jet';
select * from f where x = 'ab';
select * from f where x <> 'ab';
select * from f where x < 'ab';
select * from f where x <= 'ab';
select * from f where x > 'ab';
select * from f where x >= 'ab';
select * from f order by x;
-- test varchar(): insert w/ boundary cases
create table ff (x varchar(5));
insert into ff values ('a');
insert into ff values ('zoo');
insert into ff values ('jet');
insert into ff values ('abc');
insert into ff values ('');
insert into ff values ('a c');
insert into ff values ('abxyzxyz');
select * from ff;
select * from ff where x = 'jet';
select * from ff where x <> 'jet';
select * from ff where x < 'jet';
select * from ff where x <= 'jet';
select * from ff where x > 'jet';
select * from ff where x >= 'jet';
select * from ff where x = 'ab';
select * from ff where x <> 'ab';
select * from ff where x < 'ab';
select * from ff where x <= 'ab';
select * from ff where x > 'ab';
select * from ff where x >= 'ab';
select * from ff order by x using >;

create index f_ind on f using btree (x bpchar_ops);
create index ff_ind on ff using btree (x varchar_ops);

drop table f;
drop table ff;
