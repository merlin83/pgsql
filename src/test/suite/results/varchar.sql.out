QUERY: create table f (x char(5));
QUERY: insert into f values ('zoo');
QUERY: insert into f values ('a');
QUERY: insert into f values ('jet');
QUERY: insert into f values ('abc');
QUERY: insert into f values ('');
QUERY: insert into f values ('a c');
QUERY: insert into f values ('abxyzxyz');
QUERY: select * from f;
x      
------ 
zoo    
a      
jet    
abc    
       
a c    
abxyz  
QUERY: select * from f where x = 'jet';
x      
------ 
jet    
QUERY: select * from f where x <> 'jet';
x      
------ 
zoo    
a      
abc    
       
a c    
abxyz  
QUERY: select * from f where x < 'jet';
x      
------ 
a      
abc    
       
a c    
abxyz  
QUERY: select * from f where x <= 'jet';
x      
------ 
a      
jet    
abc    
       
a c    
abxyz  
QUERY: select * from f where x > 'jet';
x      
------ 
zoo    
QUERY: select * from f where x >= 'jet';
x      
------ 
zoo    
jet    
       
QUERY: select * from f where x = 'ab';
x  
-- 
QUERY: select * from f where x <> 'ab';
x      
------ 
zoo    
a      
jet    
abc    
       
a c    
abxyz  
QUERY: select * from f where x < 'ab';
x      
------ 
a      
       
a c    
QUERY: select * from f where x <= 'ab';
x      
------ 
a      
abc    
       
a c    
abxyz  
QUERY: select * from f where x > 'ab';
x      
------ 
zoo    
jet    
abc    
abxyz  
QUERY: select * from f where x >= 'ab';
x      
------ 
zoo    
a      
jet    
abc    
       
abxyz  
QUERY: select * from f order by x;
x      
------ 
       
a      
a c    
abc    
abxyz  
jet    
zoo    
QUERY: create table ff (x varchar(5));
QUERY: insert into ff values ('a');
QUERY: insert into ff values ('zoo');
QUERY: insert into ff values ('jet');
QUERY: insert into ff values ('abc');
QUERY: insert into ff values ('');
QUERY: insert into ff values ('a c');
QUERY: insert into ff values ('abxyzxyz');
QUERY: select * from ff;
x      
------ 
a      
zoo    
jet    
abc    
       
a c    
abxyz  
QUERY: select * from ff where x = 'jet';
x    
---- 
jet  
QUERY: select * from ff where x <> 'jet';
x      
------ 
a      
zoo    
abc    
       
a c    
abxyz  
QUERY: select * from ff where x < 'jet';
x      
------ 
a      
abc    
       
a c    
abxyz  
QUERY: select * from ff where x <= 'jet';
x      
------ 
a      
jet    
abc    
       
a c    
abxyz  
QUERY: select * from ff where x > 'jet';
x    
---- 
zoo  
QUERY: select * from ff where x >= 'jet';
x    
---- 
zoo  
jet  
     
QUERY: select * from ff where x = 'ab';
x  
-- 
QUERY: select * from ff where x <> 'ab';
x      
------ 
a      
zoo    
jet    
abc    
       
a c    
abxyz  
QUERY: select * from ff where x < 'ab';
x    
---- 
a    
     
a c  
QUERY: select * from ff where x <= 'ab';
x      
------ 
a      
abc    
       
a c    
abxyz  
QUERY: select * from ff where x > 'ab';
x      
------ 
zoo    
jet    
abc    
abxyz  
QUERY: select * from ff where x >= 'ab';
x      
------ 
a      
zoo    
jet    
abc    
       
abxyz  
QUERY: select * from ff order by x using >;
x      
------ 
zoo    
jet    
abxyz  
abc    
a c    
a      
       
QUERY: create index f_ind on f using btree (x bpchar_ops);
QUERY: create index ff_ind on ff using btree (x varchar_ops);
QUERY: drop table f;
QUERY: drop table ff;
